lib/serverless/aws-lambda.js

```diff
@@ -397,7 +399,7 @@ function setWebResponse(transaction, response) {
       responseCode
     )
 
-    const segment = transaction.agent.tracer.getSegment()
+    const segment = transaction.baseSegment
```

Comment:
Curious why baseSegment is necessary here? Can we leave a comment to indicate why?

I'm assuming this changes the behavior of which span the status code attribute is assigned to. What span was it on before and which one is it on now? Or is this just a more convenient way to grab the same one?

Reply:
> Curious why baseSegment is necessary here? Can we leave a comment to indicate why?
> 
> I'm assuming this changes the behavior of which span the status code attribute is assigned to. What span was it on before and which one is it on now? Or is this just a more convenient way to grab the same one?

When testing an async invoked lambda we did not have a current active segment at this point in the code and I noticed that in the synchronous tests the `http.statusCode` was added to the base segment. Granted, these are simple lambda operations. Should I check if we have a current segment first and if not, add it to base segement?

Reply:
OK. So in theory, these will be the same segment but in the case of the async version... we've lost state in the async hop.

I think it is fine to comment as such. I'm a little surprised we are losing state. Maybe we've already set back to the starting state at this point.